x-pack/filebeat/input/awss3/s3_input.go

```diff
@@ -233,10 +235,14 @@ func (in *s3PollerInput) readerLoop(ctx context.Context, workChan chan<- state)
 		in.metrics.s3ObjectsListedTotal.Add(uint64(totListedObjects))
 		for _, object := range page.Contents {
 			state := newState(bucketName, *object.Key, *object.ETag, *object.LastModified)
-			knownStateIDSlice = append(knownStateIDSlice, state.ID())
+			if !applyFilter(in.log, state) {
```

Comment:
I don't know if `applyFilter` is very clear to what is happening here. It seems to be a condition to see if the state can be processed or not.

Reply:
I think the naming is confusing here. I renamed to `isStateValid` which convey what it does. Hope this helps.